### ISSUE TITLE
DustCollapse point-mass test fails with too many subcycles

### ISSUE THREAD

**Issue by yut23** [AMReX-Astro/Castro#1979]
gravity_tests/DustCollapse/inputs_pointmass fails at the initial timestep, both in 1D and 2D:

```
$ ./Castro2d.gnu.ex inputs_pointmass castro.v=1
AMReX (21.09-17-g3dc694a754d4) initialized

Starting run at 02:08:41 UTC on 2021-09-10.
Successfully read inputs file ... 

Castro git describe: 21.09-8-gdeeb8c98b
AMReX git describe: 21.09-17-g3dc694a75
Microphysics git describe: 21.09-1-g744e8e6d

reading extern runtime parameters ...
3 Species: 
C12  O16  Mg24  
Successfully read inputs file ... 
Castro::numpts_1d at level  0 is 53
Setting the gravity type to ConstantGrav
Initializing the data at level 0
Done initializing the level 0 data 
...estimated hydro-limited timestep at level 0: 7.745975605
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 7.745975605


TIME= 0 MASS        = 1.33153263e+33
TIME= 0 XMOM        = 0
TIME= 0 YMOM        = 0
TIME= 0 ZMOM        = 0
TIME= 0 ANG MOM X   = 0
TIME= 0 ANG MOM Y   = 0
TIME= 0 ANG MOM Z   = 0
TIME= 0 RHO*e       = 1.23533685e+35
TIME= 0 RHO*K       = 0
TIME= 0 RHO*E       = 1.23533685e+35
TIME= 0 RHO*PHI     = 0
TIME= 0 TOTAL ENERGY= 1.23533685e+35
INITIAL GRIDS 
  Level 0   1 grids  1024 cells  100 % of domain

CHECKPOINT: file = chk00000
checkPoint() time = 0.003293086 secs.
PLOTFILE: file = plt00000
Write plotfile time = 0.000950315  seconds

...estimated hydro-limited timestep at level 0: 7.745975605
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 7.745975605

[Level 0 step 1] ADVANCE with dt = 0.07745975605

  Beginning subcycle 1 starting at time 0 with dt = 0.07745975605
  Estimated number of subcycles remaining: 1

Castro::do_old_sources() time = 7.8468e-05

... Entering construct_ctu_hydro_source()

... Leaving construct_ctu_hydro_source()

Castro::construct_ctu_hydro_source() time = 0.001812809

Castro::do_new_sources() time = 0.000115434

...estimated hydro-limited timestep at level 0: 0.0001462124365
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 0.0001462124365

Advance was unsuccessful with reason: timestep validity check failed; proceeding to a retry.

  Subcycle completed


  Timestep 0.07745975605 rejected at level 0.
  Performing a retry, with subcycled timesteps of maximum length dt = 0.03872987803


  Beginning subcycle 1 starting at time 0 with dt = 0.03872987803
  Estimated number of subcycles remaining: 2

Castro::do_old_sources() time = 7.1822e-05

... Entering construct_ctu_hydro_source()

... Leaving construct_ctu_hydro_source()

Castro::construct_ctu_hydro_source() time = 0.001832998

Castro::do_new_sources() time = 9.4029e-05

...estimated hydro-limited timestep at level 0: 0.001122995224
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 0.001122995224

Advance was unsuccessful with reason: timestep validity check failed; proceeding to a retry.

  Subcycle completed


  Timestep 0.03872987803 rejected at level 0.
  Performing a retry, with subcycled timesteps of maximum length dt = 0.01936493901


  Beginning subcycle 1 starting at time 0 with dt = 0.01936493901
  Estimated number of subcycles remaining: 4

Castro::do_old_sources() time = 6.0302e-05

... Entering construct_ctu_hydro_source()

... Leaving construct_ctu_hydro_source()

Castro::construct_ctu_hydro_source() time = 0.001769307

Castro::do_new_sources() time = 9.1325e-05

...estimated hydro-limited timestep at level 0: 0.007293108538
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 0.007293108538

Advance was unsuccessful with reason: timestep validity check failed; proceeding to a retry.

  Subcycle completed


  Timestep 0.01936493901 rejected at level 0.
  Performing a retry, with subcycled timesteps of maximum length dt = 0.009682469507


  Beginning subcycle 1 starting at time 0 with dt = 0.009682469507
  Estimated number of subcycles remaining: 8

Castro::do_old_sources() time = 7.0209e-05

... Entering construct_ctu_hydro_source()

... Leaving construct_ctu_hydro_source()

Castro::construct_ctu_hydro_source() time = 0.001711852

Castro::do_new_sources() time = 9.0672e-05

...estimated hydro-limited timestep at level 0: 0.01511447075
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 0.01511447075

  Subcycle completed


  Beginning subcycle 2 starting at time 0.009682469507 with dt = 0.009682469507
  Estimated number of subcycles remaining: 7

Castro::do_old_sources() time = 6.3675e-05

... Entering construct_ctu_hydro_source()

... Leaving construct_ctu_hydro_source()

Castro::construct_ctu_hydro_source() time = 0.001871733

Castro::do_new_sources() time = 9.3054e-05

...estimated hydro-limited timestep at level 0: 4.646846512e-08
Castro::estTimeStep (hydro-limited) at level 0:  estdt = 4.646846512e-08

Advance was unsuccessful with reason: timestep validity check failed; proceeding to a retry.

  Subcycle completed


  Timestep 0.009682469507 rejected at level 0.
  Performing a retry, with subcycled timesteps of maximum length dt = 0.004841234753


  The subcycle mechanism requested 14 subcycled timesteps, which is larger than the maximum of 10.
  If you would like to override this, increase the parameter castro.max_subcycles.
amrex::Abort::0::Error: too many subcycles. !!!
SIGABRT
See Backtrace.0 file for details

```
</details>

I did some investigating, and the test stopped working after #844. As I understand it, the initial timestep (adjusted by init_shrink) is too long, so we now reject it and subcycle with a smaller timestep. However, this is limited by max_subcycles, which ultimately means that the initial timestep has to be pretty close to the hydro-limited timestep.

Based on this, I got it to work by tweaking init_shrink, but a more helpful error message would be much better. Alternatively, would it make sense to allow more retries on the first step if the initial guess for the timestep is way off, like it is here?

**Comment by zingale** [comment]
perhaps unrelated, we get a bounds error if we run with `castro.do_grav=0`

**Comment by zingale** [comment]
also curiously, even in 2-d, there is a non-zero z gravity.

```
 grav_x                           -45412723335          -2664174.2604
 grav_y                           -45412723335          -2664174.2604
 grav_z                           -90825446669           -1885990.904
```

**Comment by maxpkatz** [comment]
The new predicted timestep of 0.00015 is indeed quite small compared to init_shrink * CFL. This suggests there's a possible problem with this setup, it's usually not great if init_shrink needs to be 0.001 for the code to not crash.

**Comment by zingale** [comment]
the z acceleration was fixed in #1982

**Comment by zingale** [comment]
the issue here is that the ambient pressure, `problem.p0` is simply too low for this problem, and the collapse is just happening too fast for the code to handle.  Increasing it from 1.e8 to 1.e12 makes the code happy.
